### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ib-steffen**

### CODE REVIEW

**Comment by sept-en** [comment]
@ib-steffen Hey!
I think this pull request is ready to merge into the master.

Could you, please, make some review on it and either merge it or provide any your thoughts about it!

Best regards,
Kirill Abramov.

File: deploy/infrabox/values.yaml [hunk 1]
```diff
@@ -146,21 +146,21 @@ static:
...
-    api:
-        # name of the dashboard-api image in your general.docker_registry
-        # image: dashboard-api
-
-        # Tag
-        tag: latest
-
-        log:
-            level: info
+#dashboard:
```
Comment (ib-steffen) [inline deploy/infrabox/values.yaml]: delete instead of comment

File: deploy/install.py [hunk 2]
```diff
@@ -347,9 +347,9 @@ def setup_github(self):
 
         self.create_secret("infrabox-github", self.args.general_system_namespace, secret)
 
-    def setup_dashboard(self):
-        self.set('dashboard.api.tag', self.args.version)
-        self.set('dashboard.url', self.args.root_url)
+    #def setup_dashboard(self):
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 3]
```diff
@@ -429,7 +429,7 @@ def main(self):
         self.setup_scheduler()
         self.setup_gerrit()
         self.setup_github()
-        self.setup_dashboard()
+        #self.setup_dashboard()
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 4]
```diff
@@ -454,16 +454,16 @@ def __init__(self, args):
...
-        self.config.append('services.dashboard-api.environment', ['INFRABOX_ROOT_URL=%s' % self.args.root_url])
-
-        self.config.append('services.dashboard-api.volumes', [
-            '%s:/var/run/secrets/infrabox.net/rsa/id_rsa' % os.path.join(self.args.o, 'id_rsa'),
-            '%s:/var/run/secrets/infrabox.net/rsa/id_rsa.pub' % os.path.join(self.args.o, 'id_rsa.pub'),
-        ])
-
-        if self.args.gerrit_enabled:
-            self.config.append('services.dashboard-api.environment', self.get_gerrit_env())
+    #def setup_dashboard(self):
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 5]
```diff
@@ -601,7 +601,7 @@ def setup_ldap(self):
                 "INFRABOX_ACCOUNT_SIGNUP_ENABLED=true"
             ]
 
-        self.config.append('services.dashboard-api.environment', env)
+        #self.config.append('services.dashboard-api.environment', env)
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 6]
```diff
@@ -638,10 +638,10 @@ def setup_database(self):
 
             self.config.append('services.docker-registry-auth.links', ['postgres'])
             self.config.append('services.scheduler.links', ['postgres'])
-            self.config.append('services.dashboard-api.links', ['postgres'])
+            #self.config.append('services.dashboard-api.links', ['postgres'])
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 7]
```diff
@@ -638,10 +638,10 @@ def setup_database(self):
...
             self.config.append('services.api.links', ['postgres'])
 
-        self.config.append('services.dashboard-api.environment', env)
+        #self.config.append('services.dashboard-api.environment', env)
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: deploy/install.py [hunk 8]
```diff
@@ -661,7 +661,7 @@ def main(self):
         self.setup_database()
         self.setup_docker_registry()
         self.setup_ldap()
-        self.setup_dashboard()
+        #self.setup_dashboard()
```
Comment (ib-steffen) [inline deploy/install.py]: delete

File: ib.py [hunk 9]
```diff
@@ -207,6 +224,14 @@ def main():
     services_kill_parser.set_defaults(func=services_kill)
     services_kill_parser.add_argument("service_name", nargs="?", type=str, help="Service name")
 
+    # Github changelog
+    changelog = cmd_parser.add_parser('changelog')
+    sub_changelog = changelog.add_subparsers()
+
+    create_changelog_parser = sub_changelog.add_parser('create')
+    create_changelog_parser.set_defaults(func=changelog_create)
+    create_changelog_parser.add_argument("--token", default='')
```
Comment (ib-steffen) [inline ib.py]: remove default, set required=False

File: ib.py [hunk 10]
```diff
@@ -170,6 +170,23 @@ def services_kill(args):
         print "Unknown service"
         sys.exit(1)
 
+def changelog_create(args):
+    # Build docker container
```
Comment (ib-steffen) [inline ib.py]: In the doc above  you write to set the env  var CHANGELOG_GITHUB_TOKEN, but I don't see any reference in the code. is it used?

File: src/api/handlers/projects/secrets.py [hunk 11]
```diff
@@ -16,9 +16,12 @@
     'value': fields.String(required=True),
 })
 
-@ns.route('/<project_id>/secrets')
+
+@ns.route('/<project_id>/secrets/')
 class Secrets(Resource):
 
+    name_pattern = re.compile('^[a-zA-Z0-9_]+$')
```
Comment (ib-steffen) [inline src/api/handlers/projects/secrets.py]: does the pattern option in the secret_model not work?

File: src/dashboard_api/__init__.py [hunk 12]
```diff
@@ -1,189 +0,0 @@
...
-
-from pyinfraboxutils import get_env, print_stackdriver, get_logger
-from pyinfraboxutils.ibrestplus import app, api
-from pyinfraboxutils.ibflask import get_token, is_collaborator
-from pyinfraboxutils import dbpool
-
-# TODO: do it the same way in api
-import dashboard_api.handlers
-
-import dashboard_api.listeners.job
```
Comment (ib-steffen) [inline src/dashboard_api/__init__.py]: did you move the listeners?

File: src/dashboard_api/__init__.py [hunk 13]
```diff
@@ -1,189 +0,0 @@
...
-        try:
-            get_token()
-        except:
-            logger.debug('disconnecting connection')
-            return False
-
-        return True
-
-    @sio.on('listen:jobs')
-    def __listen_build(project_id):
```
Comment (ib-steffen) [inline src/dashboard_api/__init__.py]: did you move the socket io handlers?

File: src/dashboard_api/__init__.py (hunk 12 above)
Comment (sept-en) [reply src/dashboard_api/__init__.py]: Yes

File: src/dashboard_api/__init__.py (hunk 13 above)
Comment (sept-en) [reply src/dashboard_api/__init__.py]: Yes

**Comment by sept-en** [comment]
Hi!
Thanks for the review!
I guess I have fixed all things you have described. Could you check it, please?

Best regards,
Kirill Abramov.

File: src/dashboard_api/__init__.py (hunk 13 above)
Comment (ib-steffen) [reply src/dashboard_api/__init__.py]: I don't see it. And be carefule the implementations are different on api and dashboard_api

**Pull request merged**